lib/eslint/eslint.js

```diff
@@ -395,6 +395,10 @@ async function calculateConfigArray(eslint, {
         const fileConfig = await loadFlatConfigFile(configFilePath);
 
         if (Array.isArray(fileConfig)) {
+            if (fileConfig.includes(void 0)) {
+                throw new Error("You have included `undefined` in your array of configs - this is commonly the result of trying to use a config from a plugin that does not exist; make sure you have not typo'd the name!");
+            }
```

Comment:
This function only loads configs for the `ESLint` class, not for `Linter` or `RuleTester`.

Maybe it would be better to add this check in [`@humanwhocodes/config-array`](https://github.com/humanwhocodes/config-array) in [`normalize`](https://github.com/humanwhocodes/config-array/blob/3dabb4db2072a86d9b567205626ba0ed537bea2f/src/config-array.js#L171-L224) and [`normalizeSync`](https://github.com/humanwhocodes/config-array/blob/3dabb4db2072a86d9b567205626ba0ed537bea2f/src/config-array.js#L226-L270), where it will run for all usages, and to make the error message more generic. What do you think @nzakas?

Reply:
I don't mind that, but I figured the error message might be ESLint specific meaning it shouldn't belong in `config-array`; I don't think it's a big deal for `RuleTester` not to do this (since that is usually used for unit type tests and can easily be verified with tools like TypeScript) but I could see it being more useful for `Linter`

Reply:
So I think what we want to do here is throw an error from `config-array` during `normalize` and `normalizeSync` that includes the index of the offending item (calling [`rethrowConfigError()`](https://github.com/humanwhocodes/config-array/blob/c1e2b9eb0c345287c6a1e1ac37d917e7536da8f1/src/config-array.js#L100). Then in ESLint we can catch that error and display an ESLint-specific message.

Reply:
Yup this is still on my radar, just been a bit busy - I hope to put something up in the next week or so, but feel free to make the pr if you want it sooner

Reply:
I'll take this.